pipeline-model-definition/src/main/resources/org/jenkinsci/plugins/pipeline/modeldefinition/generator/DirectiveGenerator/symbol-hetero-list.jelly

```diff
@@ -0,0 +1,146 @@
+    </st:attribute>
+    <st:attribute name="oneEach">
+      If true, only allow up to one instance per descriptor.
+    </st:attribute>
+    <st:attribute name="menuAlign">
+      Menu alignment against the button. Defaults to tl-bl
+    </st:attribute>
+    <st:attribute name="honorOrder">
+      If true, insert new addition by default to their 'desired' location, which
+      is the order induced by the descriptors.
+    </st:attribute>
+    <st:attribute name="capture">
+      Config fragments from descriptors are rendered lazily by default, which means
+      variables seen in the caller aren't visible to them. This attribute allows you
+      to nominate additional variables and their values to be captured for descriptors.
+    </st:attribute>
+  </st:documentation>
+  <d:taglib uri="local">
+    <d:tag name="body">
+      <table style="width:100%">
+        <j:set var="help" value="${descriptor.helpFile}" />
+        <j:if test="${hasHeader}">
+          <tr>
+            <td colspan="3">
+              <div class="dd-handle">
+                <b>${descriptor.displayName}</b>
+              </div>
+            </td>
+            <f:helpLink url="${help}"/>
+          </tr>
+          <!-- TODO: help support is unintuitive; people should be able to see help from drop-down menu -->
+          <j:if test="${help!=null}">
+            <f:helpArea />
+          </j:if>
+        </j:if>
+
+        <d:invokeBody/>
+
+        <f:class-entry descriptor="${descriptor}" />
+        <f:block>
+          <div align="right">
+            <f:repeatableDeleteButton value="${attrs.deleteCaption}" />
+          </div>
+        </f:block>
+      </table>
+    </d:tag>
+  </d:taglib>
+
+  <st:adjunct includes="lib.form.hetero-list.hetero-list"/>
+
+  <j:set var="targetType" value="${attrs.targetType?:it.class}"/>
+  <div class="hetero-list-container ${hasHeader?'with-drag-drop':''} ${attrs.oneEach?'one-each':''} ${attrs.honorOrder?'honor-order':''}">
+    <!-- display existing items -->
+    <j:forEach var="i" items="${attrs.items}">
+      <j:set var="descriptor" value="${i.descriptor}" />
+      <j:set var="instance" value="${i}" />
+      <div name="${attrs.name}" class="repeated-chunk" descriptorId="${descriptor.id}">
+        <local:body deleteCaption="${attrs.deleteCaption}">
+          <st:include from="${descriptor}" page="${descriptor.configPage}" optional="true" />
+        </local:body>
+      </div>
+    </j:forEach>
+
+    <div class="repeatable-insertion-point" />
+
+    <div class="prototypes to-be-removed">
+      <!-- render one prototype for each type -->
+      <j:set var="instance" value="${null}" />
+      <j:set var="descriptors" value="${h.filterDescriptors(it,attrs.descriptors)}" />
+      <j:forEach var="descriptor" items="${descriptors}" varStatus="loop">
+        <j:invokeStatic var="symbol" className="org.jenkinsci.plugins.pipeline.modeldefinition.generator.DirectiveGenerator" method="getSymbolForDescriptor">
```

Comment:
is `invokeStatic` working now? It used to be broken, or maybe that was just for static fields?

Reply:
Well, it is working - there are other usages of it elsewhere in Jenkins, and it definitely works here.